src/EditorFeatures/Core/Implementation/InlineRename/InlineRenameSession.OpenTextBufferManager.cs

```diff
@@ -355,6 +367,79 @@ internal void ApplyConflictResolutionEdits(IInlineRenameReplacementInfo conflict
                     // so they get classified/tagged correctly in the editor.
                     _conflictResolutionRenameTrackingSpans.Clear();
 
+                    var documentReplacements = documents
+                        .Select(document => (document, conflictResolution.GetReplacements(document.Id).Where(r => GetRenameSpanKind(r.Kind) != RenameSpanKind.None).ToImmutableArray()))
+                        .ToImmutableArray();
+
+                    var firstDocumentReplacements = documentReplacements.FirstOrDefault(d => !d.Item2.IsEmpty);
+                    var bufferContainsLinkedDocuments = documentReplacements.Length > 1 && firstDocumentReplacements.document != null;
+                    if (bufferContainsLinkedDocuments)
+                    {
+                        // Avoid a slow merge process for the case where the changes for all linked documents
```

Comment:
⚠️ This is a very important part of this improvement, since it avoids an *extremely* costly series of diffs between linked documents. It is by far the primary improvement to GC allocation rate. Please review the details carefully though.